lib/internal/errors.js

```diff
@@ -389,7 +407,110 @@ function uvException(ctx) {
   return err;
 }
 
+/**
+ * This used to be util._errnoException().
+ *
+ * @param {number} err - A libuv error number
+ * @param {string} syscall
+ * @param {string} [original]
+ * @returns {Error}
+ */
+function errnoException(err, syscall, original) {
+  // TODO(joyeecheung): We have to use the type-checked
+  // getSystemErrorName(err) to guard against invalid arguments from users.
+  // This can be replaced with [ code ] = errmap.get(err) when this method
+  // is no longer exposed to user land.
+  const code = lazyUtil().getSystemErrorName(err);
+  const message = original ?
+    `${syscall} ${code} ${original}` : `${syscall} ${code}`;
+
+  const ex = new Error(message);
+  // TODO(joyeecheung): errno is supposed to err, like in uvException
+  ex.code = ex.errno = code;
```

Comment:
In general, errors thrown by `net` and `dns` presents this weird behavior (both `err.code` and `err.errno` are the string name of the error), while errors thrown by `fs` have a numeric `err.errno` and `err.code`, which is more correct IMO (otherwise why bother duplicating the info in another property?) We might want to revisit this and see if how big the breakage would be if we make `err.errno` numeric.